@babel/types TypeScript declaration file has some errors above v7.8.0

## Bug Report

When I upgrade `@babel/types` to `7.8.0`, the following TypeScript source files had some compilation errors:

```typescript
import {
  arrowFunctionExpression,
  identifier,
  blockStatement,
  returnStatement,
  program,
  variableDeclaration,
  variableDeclarator,
  stringLiteral,
} from '@babel/types';

// error TS2554: Expected 4-7 arguments, but got 2.
arrowFunctionExpression(
  [identifier('arg')],
  blockStatement([returnStatement(identifier('arg'))]),
);

//  error TS2554: Expected 5 arguments, but got 1.
program([
  variableDeclaration('const', [
    variableDeclarator(identifier('foo'), stringLiteral('bar')),
  ]),
]);

```

I followed the error message and fixed it like this:

```typescript
arrowFunctionExpression(
  [identifier('arg')],
  blockStatement([returnStatement(identifier('arg'))]),
  false,
  false,
);

program(
  [
    variableDeclaration('const', [
      variableDeclarator(identifier('foo'), stringLiteral('bar')),
    ]),
  ],
  undefined,
  undefined,
  undefined,
  '',
);
```

Typescript compilation was successful，But when I use `node` to execute the compiled JS file，`@babel/types` threw errors:

```textplain
Error: ArrowFunctionExpression: Too many arguments passed. Received 4 but can receive no more than 3

Error: Program: Too many arguments passed. Received 5 but can receive no more than 4
```

**tsconfig.json**

```json
{
  "compilerOptions": {
    "target": "es5",
    "module": "commonjs",
    "strict": true,
    "esModuleInterop": true
  },
}
```

**Environment**

```
  System:
    OS: macOS Mojave 10.14.6
  Binaries:
    Node: 12.14.0 - /usr/local/bin/node
    Yarn: 1.21.1 - /usr/local/bin/yarn
    npm: 6.13.4 - /usr/local/bin/npm
  npmPackages:
    @babel/types: ^7.8.0 => 7.8.0 
```

--- a/packages/babel-types/scripts/generators/flow.js
+++ b/packages/babel-types/scripts/generators/flow.js
@@ -54,6 +54,7 @@ for (const type in t.NODE_FIELDS) {
 
   const struct = ['type: "' + type + '";'];
   const args = [];
+  const builderNames = t.BUILDER_KEYS[type];
 
   Object.keys(t.NODE_FIELDS[type])
     .sort((fieldA, fieldB) => {
@@ -80,8 +81,9 @@ for (const type in t.NODE_FIELDS) {
       if (typeAnnotation) {
         suffix += ": " + typeAnnotation;
       }
-
-      args.push(t.toBindingIdentifierName(fieldName) + suffix);
+      if (builderNames.includes(fieldName)) {
+        args.push(t.toBindingIdentifierName(fieldName) + suffix);
+      }
 
       if (t.isValidIdentifier(fieldName)) {
         struct.push(fieldName + suffix + ";");

--- a/packages/babel-types/scripts/generators/typescript.js
+++ b/packages/babel-types/scripts/generators/typescript.js
@@ -56,6 +56,7 @@ const lines = [];
 for (const type in t.NODE_FIELDS) {
   const fields = t.NODE_FIELDS[type];
   const fieldNames = sortFieldNames(Object.keys(t.NODE_FIELDS[type]), type);
+  const builderNames = t.BUILDER_KEYS[type];
 
   const struct = ['type: "' + type + '";'];
   const args = [];
@@ -75,18 +76,20 @@ for (const type in t.NODE_FIELDS) {
       typeAnnotation += " | null";
     }
 
-    if (areAllRemainingFieldsNullable(fieldName, fieldNames, fields)) {
-      args.push(
-        `${t.toBindingIdentifierName(fieldName)}${
-          isNullable(field) ? "?:" : ":"
-        } ${typeAnnotation}`
-      );
-    } else {
-      args.push(
-        `${t.toBindingIdentifierName(fieldName)}: ${typeAnnotation}${
-          isNullable(field) ? " | undefined" : ""
-        }`
-      );
+    if (builderNames.includes(fieldName)) {
+      if (areAllRemainingFieldsNullable(fieldName, builderNames, fields)) {
+        args.push(
+          `${t.toBindingIdentifierName(fieldName)}${
+            isNullable(field) ? "?:" : ":"
+          } ${typeAnnotation}`
+        );
+      } else {
+        args.push(
+          `${t.toBindingIdentifierName(fieldName)}: ${typeAnnotation}${
+            isNullable(field) ? " | undefined" : ""
+          }`
+        );
+      }
     }
 
     const alphaNumeric = /^\w+$/;

